packages/Tag/src/Tag.js

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import PropTypes from "prop-types";
+import * as sc from "./Tag.styles";
+import * as types from "./types";
+
+export default function Tag(props) {
+  const { children, onRemove, size, onClick, a11yText, avatar, isDisabled, ...moreProps } = props;
+
+  const handleRemove = e => {
+    onRemove();
+    e.stopPropagation();
+  };
+
+  const content = (
+    <>
+      {avatar}
+      <sc.Ellipsis size={size}>{children}</sc.Ellipsis>
+      {onRemove ? (
+        <sc.Delete isDisabled={isDisabled} size={size} data-pka-anchor="tag.remove" onClick={handleRemove} />
+      ) : null}
+    </>
+  );
+
+  if (onClick) {
+    return (
+      <sc.RawButtonTag
+        a11yText={a11yText}
+        isDisabled={isDisabled}
+        onClick={onClick}
+        size={size}
+        {...moreProps}
+        data-pka-anchor="tag"
+      >
+        {content}
+      </sc.RawButtonTag>
+    );
+  }
+
+  return (
+    <sc.Tag data-pka-anchor="tag" size={size} {...moreProps}>
+      {content}
+    </sc.Tag>
+  );
+}
+
+Tag.types = {
+  size: types.sizes,
+  theme: types.themes,
+  severityTheme: types.severityThemes,
+};
+
+const propTypes = {
+  /** used in aria-tag on the root element */
+  a11yText: PropTypes.string,
+  /** Can pass a avatar to be displayed to the left of the tag content */
+  avatar: PropTypes.node,
+  /** Can pass a custom border color */
+  borderColor: PropTypes.string,
```

Comment:
I think the best part about having a `theme` prop was the ability to remove the unrestricted `borderColor` prop and roll that into it.

Reply:
I played about with it and because we need to have the height of the Tag be adjusted if there is a border being applied it started to get messy applying border as part of the theme as well....

https://github.com/acl-services/paprika/pull/929/files#diff-c6887f020b1e2c81ec422eb1bcf30718a2e7104a195bb691e717bc16d700892dR98

Providing the `borderColor` prop still gives flexibility to the consumer to apply a `borderColor` when they like (which affects the `height` and `padding` on the Tag) so I think is an enhancement. I think for now, this works but perhaps can revisit how the border can still be applied via the theme later.

Reply:
Got it to work! I did try with border-box before but kept running into issues 🤷‍♂️